src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/EmployeeController.php

```diff
@@ -216,32 +223,194 @@ public function bulkDeleteAction(Request $request)
+
+        if (!$formAccessChecker->canAccessEditFormFor($employeeId)) {
+            $this->addFlash(
+                'error',
+                $this->trans('You cannot edit the SuperAdmin profile.', 'Admin.Advparameters.Notification')
+            );
+
+            return $this->redirectToRoute('admin_employees_index');
+        }
+
+        $isRestrictedAccess = $formAccessChecker->isRestrictedAccess((int) $employeeId);
+        $canAccessAddonsConnect = $formAccessChecker->canAccessAddonsConnect();
 
-        $errorMessages = [
-            InvalidEmployeeIdException::class => $this->trans('The object cannot be loaded (the identifier is missing or invalid)', 'Admin.Notifications.Error'),
-            EmployeeNotFoundException::class => $this->trans('The object cannot be loaded (or found)', 'Admin.Notifications.Error'),
+        $employeeForm = $this->getEmployeeFormBuilder()->getFormFor((int) $employeeId, [], [
+            'is_restricted_access' => $isRestrictedAccess,
+            'is_for_editing' => true,
+            'show_addons_connect_button' => $canAccessAddonsConnect,
+        ]);
+
+        try {
+            $employeeForm->handleRequest($request);
+            $result = $this->getEmployeeFormHandler()->handleFor((int) $employeeId, $employeeForm);
+
+            if (null !== $result->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_employees_index');
+            }
+        } catch (EmployeeException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        $editableEmployee = $this->getQueryBus()->handle(new GetEmployeeForEditing((int) $employeeId));
+
+        $templateVars = [
+            'employeeForm' => $employeeForm->createView(),
+            'isRestrictedAccess' => $isRestrictedAccess,
+            'showAddonsConnectButton' => $canAccessAddonsConnect,
+            'editableEmployee' => $editableEmployee,
+        ];
+
+        return $this->render(
+            '@PrestaShop/Admin/Configure/AdvancedParameters/Employee/edit.html.twig',
+            $templateVars + $this->getFormTemplateVariables($request)
+        );
+    }
+
+    /**
+     * Change navigation status for employee.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function toggleNavigationAction(Request $request)
+    {
+        $navigationHandler = $this->get('prestashop.adapter.employee.navigation_handler');
+        $navigationHandler->toggleNavigation($request->request->getBoolean('shouldCollapse'));
+
+        return new Response('', Response::HTTP_NO_CONTENT);
+    }
+
+    /**
+     * Change employee form language.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function changeFormLanguageAction(Request $request)
```

Comment:
The need for security is not related to the content of the controller action. Right now it modifies data into the cookie so indeed this information is not secured, but
- we might in the future modify this controller action to perform other operations such as database modification, event dispatching
- if a module hooks himself on this action it might do more things that require security

In fact, almost all Controller actions in the BO must be secured by default. Only some rare cases such as ajax GET actions which provide data accross all BO can be open.

Reply:
I see, but which permissions should be checked when performing this action?

Reply:
`@AdminSecurity("is_granted(['update'], request.get('_legacy_controller'))")` will do 👍